src/software/Localization/main_SfM_Localization_MultipleImages.cpp

```diff
@@ -0,0 +1,341 @@
+      }
+      else // load already existing regions
+      {
+        query_regions->Load(sFeat,sDesc);
+      }
+    }
+
+    std::shared_ptr<cameras::IntrinsicBase> optional_intrinsic(nullptr);
+    if (cmd.used('s') && num_initial_intrinsics == 1)
+    {
+      optional_intrinsic = std::make_shared<cameras::Pinhole_Intrinsic_Radial_K3>(
+        imageGray.Width(), imageGray.Height(),
+        ptrPinhole->focal(), ptrPinhole->principal_point()[0], ptrPinhole->principal_point()[1],
+        ptrPinhole->getParams()[3], ptrPinhole->getParams()[4], ptrPinhole->getParams()[5]);
+    }
+
+    geometry::Pose3 pose;
+    sfm::Image_Localizer_Match_Data matching_data;
+    matching_data.error_max = dMaxResidualError;
+
+    // Build the view corresponding to the image
+    View v(*iter_image, views.size(), views.size(), views.size(), imageGray.Width(), imageGray.Height());
+
+    // Try to localize the image in the database thanks to its regions
+    if (!localizer.Localize(
+      Pair(imageGray.Width(), imageGray.Height()),
+      optional_intrinsic.get(),
+      *(query_regions.get()),
+      pose,
+      &matching_data))
+    {
+      std::cerr << "Cannot locate the image " << *iter_image << std::endl;
+      v.id_intrinsic = UndefinedIndexT;
+      v.id_pose = UndefinedIndexT;
+    }
+    else
+    {
+      const bool b_new_intrinsic = (optional_intrinsic == nullptr);
+      // A valid pose has been found (try to refine it):
+      // If not intrinsic as input:
+      // init a new one from the projection matrix decomposition
+      // Else use the existing one and consider as static.
+      if (b_new_intrinsic)
+      {
+        // setup a default camera model from the found projection matrix
+        Mat3 K, R;
+        Vec3 t;
+        KRt_From_P(matching_data.projection_matrix, &K, &R, &t);
+
+        const double focal = (K(0,0) + K(1,1))/2.0;
+        const Vec2 principal_point(K(0,2), K(1,2));
+        optional_intrinsic = std::make_shared<cameras::Pinhole_Intrinsic_Radial_K3>(
+          imageGray.Width(), imageGray.Height(),
+          focal, principal_point(0), principal_point(1));
+
+      }
+      sfm::SfM_Localizer::RefinePose
+      (
+        optional_intrinsic.get(),
+        pose, matching_data,
+        true, b_new_intrinsic
+      );
+
+      vec_found_poses.push_back(pose.center());
+
+      // Add the computed intrinsic to the sfm_container
+      intrinsics[v.id_intrinsic] = optional_intrinsic;
+      
+      // Add the computed pose to the sfm_container
+      poses[v.id_pose] = pose;
+
+    }
+    
+    // Add the view to the sfm_container
+    views[v.id_view] = std::make_shared<View>(v);
+  }
+
+  GroupSharedIntrinsics(sfm_data);
+  
+  std::cout << " Total poses found : " << vec_found_poses.size() << "/" << vec_image_new.size() << endl;
+
+  // Export the found camera position in a ply.
+  const std::string out_file_name = stlplus::create_filespec(sOutDir, "found_pose_centers", "ply");
```

Comment:
You can use the existing function to do that (save all the camera positions):

FYI:
(**Save** can export scene in PLY and you can choose which information you want to export):
```
Save(
     sfm_data,
      out_file_name .c_str(),
     ESfM_Data(EXTRINSICS)));
```
If you want to save only the new found poses you can use this function:
https://github.com/openMVG/openMVG/blob/master/src/software/SfM/SfMPlyHelper.hpp#L21
```
plyHelper::exportToPly(vec_found_poses, out_file_name);
```